runeq/resources/org.py

```diff
@@ -188,11 +150,121 @@ def get_orgs(client: Optional[GraphClient] = None) -> OrgSet:
                 created_at=org_attrs.get("created_at"),
                 name=org_attrs.get("name"),
             )
-            org_set.add(org)
+            yield org
 
         # endCursor is None when there are no more pages of data
         next_cursor = membership_list.get("pageInfo", {}).get("endCursor")
         if not next_cursor:
             break
 
+
+def get_org(org_id: str, client: Optional[GraphClient] = None) -> Org:
+    """
+    Get the org with the specified ID.
+
+    Args:
+        org_id: Organization ID
+        client: If specified, this client is used to fetch metadata from the
+            API. Otherwise, the global GraphClient is used.
+
+    """
+    org_id = Org.normalize_id(org_id)
+    # The getOrg query only works for the user's currently active organization.
+    # To make this work for any org that the user is a member of, use
+    # getOrgMemberships and try to find a matching org ID. It's slow, but it
+    # works for now.
+    for org in _iter_all_orgs(client):
+        if org_id == org.id:
+            return org
+
+    raise ValueError("Org not found with id: %s" % org_id)
```

Comment:
Nit: why not use an f-string? (It looks like we're not supporting anything older than Python 3.6)
```suggestion
    raise ValueError(f"Org not found with id: f{org_id}")
```